### DISCUSSION TITLE
AppImage for releases

### DISCUSSION THREAD

**Discussion by spamatica** [muse-sequencer/muse#1195] [Team Posts]
Hi kybos, it was probably you that added an AppImage to the previous prerelease. What is needed to to create an appimage for a release?

In a sense the app images removes the need for making prereleases, since we are doing continuous releases. So maybe we are gearing up for a 4.0 release? Need to decide the scope once and for all though if there is something we really want to get in there. For instance that source code renaming business.

**Comment by kybos** [comment]
> What is needed to to create an appimage for a release?

I created a special action for it - "Create AppImage for release". It has the release tag as input parameter and the finished AppImage as output, so it's pretty easy to use.

> In a sense the app images removes the need for making prereleases, since we are doing continuous releases. So maybe we are gearing up for a 4.0 release? Need to decide the scope once and for all though if there is something we really want to get in there. For instance that source code renaming business.

Maybe some people still prefer fixed releases, expecting them to be more stable? I don't know...
As for the 4.0 release: I think it would be better to do the renaming independently, and before the final release, just in case there are problems that have to be fixed. So feel free to do it anytime.
I still have some things open that should go into 4.0, but that should be finished by this weekend, so if you want to do the release then, it's ok with me.

There are some issues that I personally find pretty critical (quite basic midi issues with the recording modes,  punch in/out etc., there are several tickets on that), where I'm a little afraid they could put users off quickly who will be trying out the new version (but I may be biased, as I myself use only the midi parts in MusE...). I know Tim was working on that but I understand he is very busy ATM, so I suspect there is nothing coming in the short term.

**Comment by spamatica** [comment]
Oh hey! 
Sadly I'm mostly in another timezone (mentally) so didn't get any further yet but felt I wanted to get started atleast.
A pre-release could happen next weekend but the real release I think needs more work than I can provide that fast..
I didn't mean that we should skip doing releases now that we have the appimages. There are still plenty of reasons for doing old-style stable releases I think :) It was specifically the pre-releases that (atleast in theory) are provided to get feedback but since the appimages are so easy to try maybe the reasons are less. Still they aren't that much work so I might just as well do one, it gives a an opportunity to go through the changelog and try to filter out a list of new features.

Yeah, the punch in/out for midi is a sore spot, but as Tim seems to be in a bit of a bad place at the moment I'd not want to add to his stress. Let's hope it gets better come spring.

**Comment by spamatica** [comment]
And really cool that it's so easy to make a release with the appimage then I don't have to think about that :)

**Comment by kybos** [comment]
> I didn't mean that we should skip doing releases now that we have the appimages.

Yes I know, I was also meaning the pre-releases only. I noticed in some tickets that people used the pre-release appimage and not the current master one, probably thinking it's more stable than the master. So yes, if you have time, pre2 would be good, there have been many fixes and new features since pre1. And we could try out the semver naming ;-).
